### DISCUSSION TITLE
currently I training glow model on 400 EPOCH. I test some sentences but that sound like robot . and my  avg_log_mle go in negative and still increasing but my avg_loss_dur stuck between 0.40-0.45 . I attach outputs of test data

### DISCUSSION THREAD

**Discussion by omkarade** [coqui-ai/TTS#2000] [General Q&A]
batch_size=32,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=1000,
    text_cleaner="hindi_cleaners",
    use_phonemes=False,
    dashboard_logger='tensorboard', 
    save_step=8000, 
    distributed_backend='nccl',
    distributed_url='tcp://localhost:54321', 
    #phoneme_language="en-us",
    #phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    print_step=250,
    print_eval=False,
    mixed_precision=True,
    audio=audio_config,
    characters=character_config,
    output_path=output_path,

glow_tts_output---https://soundcloud.com/omkr3345/glow-tts-output?si=44f2430615f24cb18eeda62649feb72e&utm_source=clipboard&utm_medium=text&utm_campaign=social_sharing

**Comment by WeberJulian** [comment]
It's normal that it sounds like a robot, you don't have any vocoder. But is it inteligible?

**Reply by omkarade** [reply]
glow_tts_output---https://soundcloud.com/omkr3345/glow-tts-output?si=44f2430615f24cb18eeda62649feb72e&utm_source=clipboard&utm_medium=text&utm_campaign=social_sharing
how make this  audio smooth?

**Reply by omkarade** [reply]
how add any vocoder?

**Reply by WeberJulian** [reply]
Can you understand what the test sentence are saying?

**Reply by omkarade** [reply]
yes

**Reply by WeberJulian** [reply]
Ok then you need to train a vocoder, I suggest fintuning hifigan

**Reply by omkarade** [reply]
pleas tell me how add any vocoder? I search in coqui documentation but not found anything regarding that

**Reply by omkarade** [reply]
how do fintuning  hifigan any guideline or steps

**Reply by WeberJulian** [reply]
https://github.com/coqui-ai/TTS/blob/dev/recipes/ljspeech/hifigan/train_hifigan.py

**Reply by omkarade** [reply]
my  avg_loss_dur stuck between 0.40-0.45 since 200 EPOCH. what should I do? I fintuning hifigan or wait for avg_loss_dur get more reduce??

 --> EVAL PERFORMANCE
     | > avg_loader_time: 0.00684 (+0.00381)
     | > avg_loss: 0.30068 (+0.01359)
     | > avg_log_mle: -0.13370 (+0.00812)
     | > avg_loss_dur: 0.43439 (+0.00548)

**Reply by WeberJulian** [reply]
If it didn't improve in 200 epochs, it won't improve at all

**Reply by omkarade** [reply]
for doing **fintuning** I just need to give glow models output path in  **TrainerArgs("glow output path")**  in [train_hifigan.py]?
correct me if I wrong.

# init model
model = GAN(config, ap)

# init the trainer and 🚀
trainer = Trainer(
    TrainerArgs(), config, output_path, model=model, train_samples=train_samples, eval_samples=eval_samples
)
trainer.fit()

**Reply by omkarade** [reply]
@WeberJulian ??????????
